### DISCUSSION TITLE
Helix golf competition! 1. Turn a list into an array of strings using as few keystrokes as possible

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#8130] [Show and tell]
Please try and beat my Helix golf solutions! You must end in Normal Mode. Let the games begin!
## Game 1

I borrowed the idea from https://stackoverflow.com/questions/27473633/how-to-follow-a-vim-golf-solution which contains the original Vim solution. I have not tried to convert it, but rather use Helix functionality. Your cursor starts on the first letter of `Hello`
```
Hello
This
Is
Helix
```
![image](https://github.com/helix-editor/helix/assets/12832280/e27c6685-2ec5-4915-8588-3e398e6b0b26)

Becomes:

`[ "Hello", "This", "Is", "Helix" ]`

![image](https://github.com/helix-editor/helix/assets/12832280/b7f3e4aa-f08c-4df5-98e0-dcfa6a7e6d53)

1. WINNING SOLUTIONS by @gabydd (19 key presses)

`%<A-s>ms"<A-,>I<backspace>, <esc>x_ms ms[`
`%<A-s>ms"(<A-,>a,<esc>Jx_ms ms[`

2. Solution 1 (20 key presses):

`%<ALT s>ms"a,<ESC>J<ALT ->;ddxms<SPACE>ms[`

## Game 2

Start with `8x`, there is imaginary white-space around the text:
```
RD 5 Gore 9775
RD 6 Gore 9776
RD 7 Gore 9777
RD 1 Great Barrier Island 0991
RD 1 Greta Valley 7387
RD 1 Greytown 5794
RD 1 Hamilton 3281
RD 2 Hamilton 3282
```
becomes:
```
9775
9776
9777
0991
7387
5794
3281
3282
```

1. Solution 1 (8 key presses):

`8x<ALT s>4hT<ENTER>d`

2. I also came up with this garbage before I got the above.:

`8x<ALT s>s\d{4}<ENTER><ALT ;>hT<ENTER>d`

**Comment by gabydd** [comment]
``%<A-s>ms"<A-,>I<backspace>, <esc>x_ms ms[`` 19 (working on a different method not sure if it will be less keys)

``%_ms ms[<A-s>b_ms"<A-,>I<backspace>, <esc>`` 20 actually, trying to improve this one (``%<A-s>ms"(<A-,>a,<esc>%ms ms[<A-s>J`` and ``%ms ms[<A-s>b_ms"<A-,>i<backspace>, <esc>J
`` 19)

Another 19 solution more inline with yours:
``%<A-s>ms"(<A-,>a,<esc>Jx_ms ms[``

**Reply by David-Else** [reply]
Nice!

**Reply by gabydd** [reply]
Thanks I found a way to do it in 18 but it ends in insert mode so I don't know if that counts

**Reply by David-Else** [reply]
Mmm... not sure if it counts, didn't really think about it. The main thing is to see how different people approach the problem to help learn new patterns. I will add it must end in normal mode to the original post.

**Reply by David-Else** [reply]
Hey @gabydd I added another in the original comment as you like golf :)

**Reply by gabydd** [reply]
``8x<A-s>by_xR`` is 8 which works even if there is no whitespace before if there is whitespace you can also change the 4h to b in your solution to make it 7

**Reply by gabydd** [reply]
Best regex based solution is probably ``%S\d+<return>$_dd`` or ``%s\d+$<return>yx_R``

**Reply by gabydd** [reply]
``7Cv3wd`` 6!

**Reply by David-Else** [reply]
`7Cv3wd` I love it, so simple and concise! I will add you to the main post...

**Reply by gabydd** [reply]
Thanks for the fun challenges

**Reply by David-Else** [reply]
I am half asleep, are you sure that last one `7Cv3wd` works and doesn't leave this? I might be asleep now actually.
```
9775
9776
9777
Barrier Island 0991
Valley 7387
5794
3281
3282
```

**Reply by gabydd** [reply]
Lol yeah it does don't know how I didn't notice

**Reply by gabydd** [reply]
The best one is 7 still ``8x<A-s>bT<return>d`` then

**Reply by plul** [reply]
No-brainer in 8:

```
8x<A-s>bhv<home>d
```

I can't get any of the ones posted above that claims to be under 8 to work. For example, this one chews a digit when I try it: `8x<A-s>bT<return>d`.

**Reply by aral** [reply]
Yeah, I haven’t been able to get it under 8 either:

```
7Cvglbhd
7CvglF d
```

**Reply by aral** [reply]
Ah, got it to 7 :)

```
7Cv<End>bhd
```

**Reply by gabydd** [reply]
Ah 7 and it actually works: ``%<A-s>_vbhd`` which also can be 6 if there is no whitespace around cause you don't need the ``_``

**Reply by gabydd** [reply]
Fun regex 7: ``%s.+ <return>d``

**Comment by pascalkuthe** [comment]
for the first challange you can get even better:

If the file does not have a trailing newline you can get down to 17:

`%<A-s>ms"<A-J>i,<esc>x_ms ms[`

If it does this  variant needs one more keypress (so 18): 

`%<A-s>ms"<A--><A-J>i,<esc>x_ms ms[`

**Reply by gabydd** [reply]
Very nice solution, looks like ``A-J`` gets seen as ``J`` on termux but using the command picker works (I was trying to use the alt version but it wasn't doing anything). Also without a trailing newline I believe this is actually 16 then because there is no need for the ``_`` after the ``x``.

**Reply by gabydd** [reply]
For no trailing newline another solution would be for 16 ``%<A-s>ms"%ms ms[<A-J>i,<esc>`` and that gives us ``%<A-s>ms"%_ms ms[<A-J>i,<esc>`` for 17 with the newline

**Comment by cd-a** [comment]
do we have any more of those? that is really fun

**Reply by James-Firth** [reply]
I also learned a lot from reading these solutions!

**Reply by nightscape** [reply]
Here's an article on Medium with a few more: 
https://circuit4us.medium.com/play-vimgolf-with-helix-editor-b8c94970fe95

**Comment by nik-rev** [comment]
Here's a website which showcases more of these examples: https://nik-rev.github.io/helix-golf/

**Comment by ghost** [comment]
Another solution for game 2 in 7 keypresses:
``8xs<space><enter>bd``
``%s_<space><enter>bd``

Without whitespace around the text it could be 6:
``%s<space><enter>bd``
